File: ICPC2025/1date/DeliveringOrders.py
K = int(input())
inventory = list(map(int, input().split()))
N = int(input())
orders = [list(map(int, input().split())) for _ in range(N)]

day = 1
while True:
    can_fulfill = True
    for i in range(N):
        order_fulfilled = True
        for j in range(K):
            if orders[i][j] > inventory[j]:
                order_fulfilled = False
                break
        if order_fulfilled:
            for j in range(K):
                inventory[j] -= orders[i][j]
        else:
            print(day, i + 1)
            exit()
    day += 1

File: ICPC2025/1date/DeliveringOrders.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int K;
    cin >> K;
    vector<long long> inventory(K);
    for (int i = 0; i < K;i++) {
        cin >> inventory[i];
    }

    int N;
    cin >> N;
    vector<vector<int>> orders(N, vector<int>(K));
    for (int i = 0; i < N;i++) {
        for (int j = 0; j < K;j++) {
            cin >> orders[i][j];
        }
    }

    int day = 1;
    while (true) {
        bool canFulfill = true;
        for (int i = 0; i < N;i++) {
            bool orderFulfilled = true;
            for (int j = 0; j < K;j++) {
                if (orders[i][j] > inventory[j]) {
                    orderFulfilled = false;
                    break;
                }
            }
            if (orderFulfilled) {
                for (int j = 0; j < K;j++) {
                    inventory[j] -= orders[i][j];
                }
            } else {
                cout << day << " " << (i + 1) << endl;
                return 0;
            }
        }
        day++;
    }

    return 0;
}